### PR TITLE
feat(preact): update to X (v10)

### LINKED ISSUE #656: Preact X
Preact X (10) is out, seems like it has fixed some issues in compatibility.

Maybe it is worth to update the version?

I can do it, if you are not against

### PR DESCRIPTION
**Pull request by askbeka**
fixes #656

### CODE REVIEW

**Comment by askbeka** [comment]
@robdodson seems like I can not provide `GITHUB_TOKEN`

File: libraries/dio/src/components.js [hunk 1]
```diff
@@ -199,7 +199,7 @@ export class ComponentWithDeclarativeEvent extends Component {
           onlowercaseevent={this.handleLowercaseEvent}
           onkebab-event={this.handleKebabEvent}
           oncamelEvent={this.handleCamelEvent}
-          onCAPSEvent={this.handleCapsEvent}
+          onCAPSevent={this.handleCapsEvent}
```
Comment (askbeka) [inline libraries/dio/src/components.js]: wrong event name with captial `E`vent

File: libraries/preact/src/components.js [hunk 2]
```diff
@@ -199,7 +199,7 @@ export class ComponentWithDeclarativeEvent extends Component {
           onlowercaseevent={this.handleLowercaseEvent}
           onkebab-event={this.handleKebabEvent}
           oncamelEvent={this.handleCamelEvent}
-          onCAPSEvent={this.handleCapsEvent}
+          onCAPSevent={this.handleCapsEvent}
```
Comment (askbeka) [inline libraries/preact/src/components.js]: wrong event name with captial `E`vent

File: libraries/react/src/components.js [hunk 3]
```diff
@@ -202,7 +202,7 @@ export class ComponentWithDeclarativeEvent extends Component {
           onlowercaseevent={this.handleLowercaseEvent}
           onkebab-event={this.handleKebabEvent}
           oncamelEvent={this.handleCamelEvent}
-          onCAPSEvent={this.handleCapsEvent}
+          onCAPSevent={this.handleCapsEvent}
```
Comment (askbeka) [inline libraries/react/src/components.js]: wrong event name with captial `E`vent

File: libraries/skate/src/components.js [hunk 4]
```diff
@@ -226,7 +226,7 @@ class ComponentWithDeclarativeEvent extends withComponent(withPreact()) {
           onlowercaseevent={this.handleLowercaseEvent}
           onkebab-event={this.handleKebabEvent}
           oncamelEvent={this.handleCamelEvent}
-          onCAPSEvent={this.handleCapsEvent}
+          onCAPSevent={this.handleCapsEvent}
```
Comment (askbeka) [inline libraries/skate/src/components.js]: wrong event name with captial `E`vent

**Comment by askbeka** [comment]
@robdodson I have found out typos in event listener properties `onCAPSEvent` should be `onCAPSevent`

If it is ok it can go with this merge request.

Otherwise I can create separate MR for that.

**Comment by robdodson** [comment]
oh good catch. I think you can add that change to this PR.

**Comment by askbeka** [comment]
@robdodson Thanks.Changes are already in PR.

According to tests Preact is passing 14/14.

But I see some test were commented I have upgraded them too,
2 tests from commented ones are not passing, they are related to properties of unregistered custom elements.

I am not sure if I need to change a [issues.json](https://github.com/webcomponents/custom-elements-everywhere/blob/master/libraries/preact/meta/issues.json) file. I need to spend more time to understand if those issues are resolved in Preact X.

**Comment by askbeka** [comment]
@robdodson Ok. Looks good enough to me.

Build is failing because of GITHUB_TOKEN env variable, I cannot set mine since I do not have rights in Travis, But I have created a [PR in my fork](https://github.com/askbeka/custom-elements-everywhere/pull/1) and it is [building ok](https://travis-ci.com/askbeka/custom-elements-everywhere/builds/133261464) there.

**Comment by developit** [comment]
Thank you so much @askbeka!

**Comment by askbeka** [comment]
@robdodson are there any blockers, something preventing merging this PR.
I can try to help.

**Comment by dntzhang** [comment]
@robdodson @askbeka  I met it too. How to solve it
https://github.com/webcomponents/custom-elements-everywhere/pull/665

**Comment by robdodson** [comment]
I pulled down the branch and made some changes to the summary.md and pushed to master.
https://github.com/webcomponents/custom-elements-everywhere/commit/b45feece29df70788e87fb9aca35c83327e887e7

This should go live shortly. Thank you for all your work @askbeka !

**Pull request closed without merging**